File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys

while True:
    edges = sorted(list(map(int, sys.stdin.readline().split())))
    
    if edges[0] == 0:
        break
    
    # 가장 먼저 삼각형을 만들 수 있느냐 없느냐 부터 판단하고 들어가자
    if edges[0] + edges[1] <= edges[2]: # 삼각형을 만들 수 없는 경우
        print("Invalid")
    
    else: # 삼각형을 만들 수 있는 경우
        if edges[0] == edges[1] and edges[1] == edges[2]:
            print("Equilateral")
    
        elif edges[0] != edges[1] and edges[1] != edges[2] and edges[0] != edges[2]:
            print("Scalene")
    
        else:
            print("Isosceles")

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <algorithm>
#include <vector>

int main(int argc, char** argv){
    while(true){
      int a, b, c;
      std::cin >> a >> b >> c;
      
      std::vector<int>lines = {a, b, c};
      std::sort(lines.begin(), lines.end());
      int max_line = lines[2];
 
      if(lines[0] == max_line && lines[1] == max_line && lines[0] == lines[1]){
          if(max_line != 0){
            std::cout << "Equilateral" <<"\n";  
          } 
          else {
              break;
          }
      }
      
      else if(max_line >= lines[0] + lines[1]){
          std::cout << "Invalid" << "\n";
      }
        
      else if(lines[0] != lines[1] && lines[0] != max_line && lines[1] != max_line){
          std::cout << "Scalene" << "\n";
      }
          
      else std::cout << "Isosceles" << "\n";
    }
    
    return 0;
}